eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnifiedMap.java

```diff
@@ -1430,6 +1430,94 @@ public void value(K key, V value)
         return builder.toString();
     }
 
+    public boolean trimToSize()
+    {
+        if (this.table.length <= this.occupied)
```

Comment:
the size calculations and later allocation look wrong. You shouldn't be comparing table.length (because in this map there is an allocation for both keys and values, resulting in a factor of 2). You should also be taking the load factor into account. You should never call allocateTable with something that is not a power of 2.

look at the init method. It computes the correct size, given a capacity. you can refactor into a new method to return the computed size:
```
        int capacity = 1;
        while (capacity < occupied)
        {
            capacity <<= 1;
        }
```
if ((capacity << 1) < table.length) then resize (by calling allocate(capacity)).